zwave-classifier.js

```diff
@@ -49,22 +53,84 @@ const QUIRKS = [
   },
   {
     // The Aeotec ZW096 says it supports the MULTILEVEL command class, but
-    // setting it acts like a no-op. We remove the level property so that
+    // setting it acts like a no-op. We remove the 'level' property so that
     // the UI doesn't see it.
     zwInfo: {
       manufacturerId: AEOTEC_MANUFACTURER_ID,
       productId: AEOTEC_ZW096_PRODUCT_ID,
     },
     excludeProperties: ['level'],
   },
+  {
+    // The Aeotec ZW100 says it supports the SENSOR_BINARY command class,
+    // but this is only true for some configurations. We use the alarm
+    // command class instead.
+    // We remove the 'on' property so that the UI doesn't see it.
+    zwInfo: {
+      manufacturerId: AEOTEC_MANUFACTURER_ID,
+      productId: AEOTEC_ZW100_PRODUCT_ID,
+    },
+    excludeProperties: ['on'],
+
+    setConfigs: [
+      // Configure motion sensor to send 'Basic Set', rather than
+      // 'Binary Sensor report'.
+      {instance: 1, index: 5, value: 1}
+    ],
+  },
 ];
 
+function quirkMatches(quirk, node) {
+  let match = true;
+  for (const id in quirk.zwInfo) {
+    if (node.zwInfo[id] !== quirk.zwInfo[id]) {
+      match = false;
+      break;
+    }
+  }
+  return match;
+}
+
+
 class ZWaveClassifier {
 
   constructor() {
   }
 
   classify(node) {
+    this.classifyInternal(node);
+
+    // Any type of device can be battery powered, so we do this check for
+    // all devices.
+    const batteryValueId = node.findValueId(COMMAND_CLASS_BATTERY, 1, 0);
+    if (batteryValueId) {
+      this.addBatteryProperty(node, batteryValueId);
+    }
+  }
+
+  classifyInternal(node) {
+    // Search through the known quirks and see if we need to apply any
+    // configurations
+    for (const quirk of QUIRKS) {
+      if (!quirk.hasOwnProperty('setConfigs')) {
+        continue;
+      }
+
+      if (quirkMatches(quirk, node)) {
+        for (const setConfig of quirk.setConfigs) {
```

Comment:
You should ensure that `quirk.hasOwnProperty('setConfigs')` here before looping, otherwise this could throw an error.

Reply:
There is a check at the top of the loop which continues if setConfigs doesn't exist.